### DISCUSSION TITLE
Zigbee direct binding

### DISCUSSION THREAD

**Discussion by andrazek** [doctor64/tuyaZigbee#31] [General]
Zigbee smart lights are gaining popularity, and since they act as routers, it's preferred to keep them powered continuously to prevent disruptions to the Zigbee mesh network. Can a custom firmware be developed for the Tuya TS0001 to support Zigbee direct binding to these smart lights? This would require exposing the 'genonoff' output cluster, which is currently not available in the official firmware. Additionally, it would be beneficial to have the option to switch to detached mode for the relay inside the switch module, but this might not be possible due to hardware limitations. However, it's still possible to directly wire a smart light, bypassing the relay of the switch module. The ability to directly bind such smart switch modules to smart lights is crucial to overcome issues when a server or Zigbee network is down for any reason.

**Comment by doctor64** [comment]
All my firmwares supports direct binding and reporting.
Unfortunately, as noted in README, Tuya device names like TS0001 are more like device classes, and cannot be used to identify hardware.
So I need at least photos, preferably links to shop where I can purchase device for development (Aliexpress preferred). If device based on Telink chip (look on board or check MAC address of device, details in README) it usually can be implemented.

**Reply by andrazek** [reply]
Here is the link (Girier):
https://a.aliexpress.com/_EJvLrSL

The starting part of their MAC is 0xa4c138 so I suppose they have Telink inside. I have either 1 gang and 2 gang modules at home. Those modules have the smallest form factor and work reliably. Support for the direct binding would mean so much for the community! How about adding an option for the decoupled mode, is it feasible?

**Reply by jbro164** [reply]
I would be keen for this as well (same use case) including decoupling the buttons and switches.  I have these https://www.zemismart.com/products/kes-606us-l1 which come with a ZS3L chip (also on aliexpress here:https://www.aliexpress.com/item/1005005396473245.html).  I have done a little bit of investigation, and looks like the input switches are connected to pins 2,3,4,5 and the output relays connected from pins 10,11,12,13 (this is from quick investigation. need to buzz out all the connections to 100% confirm).  Photos of inside the unit here
![20240429_204806](https://github.com/doctor64/tuyaZigbee/assets/74098920/148c08dc-2abe-4077-9870-29beb4ef332a)
![20240429_204833](https://github.com/doctor64/tuyaZigbee/assets/74098920/8302453d-19f2-4370-aad0-4a96cd09f9fd)
The relays are connected to the 8pin Header on a separate PCB (along with the powersupply etc)

I can help and test if needed, just need a few pointers, as not a very strong programmer.

**Reply by doctor64** [reply]
Open issue for tracking https://github.com/doctor64/tuyaZigbee/issues/33

**Reply by andrazek** [reply]
Should we open another issue for Girier switch modules which were in initial post? They are different from Zenismart switches.

**Reply by doctor64** [reply]
### JR-ZDS01 Girer mini switch
https://www.aliexpress.com/item/1005002164359835.html
already in queue. See in docs/devices-wip.md

**Reply by doctor64** [reply]
@andrazek 
also, this device looks similar
### weten 2 gang no Neutral
https://www.aliexpress.com/item/1005006089409372.html
state: arrived

**Comment by andrazek** [comment]
I don't think it's the same. Physically it looks different, I have also those from queue at home, but the form factor is a bit bigger so I prefer the other Girier modules. They are also twice as expensive than JR-ZDS01.

**Reply by doctor64** [reply]
Please provide photos of your device, if possible - board too.

**Reply by doctor64** [reply]
Sorry, and please open separate issue for this device with all possible info - links, photos, etc
